### ISSUE TITLE
liked/fav'd posts from OStatus users are shown in TL

### ISSUE THREAD

**Issue by psychohype** [friendica/friendica#2364]
I follow several OStatus users. And if one of them likes/fav's something, I get that post in my time line. This behavior is a little bit annoying. I think there should be a way to hide those posts.

**Comment by tobiasd** [comment]
I think the "Only import OStatus threads from our contacts" option is not working correctly., this may cause the false import of the faved notices.

**Comment by annando** [comment]
The OStatus thread completion is the oldest part in the whole protocol implementation - and surely needs and overhaul.

**Comment by annando** [comment]
BTW: This issue should be solved in 3.5.1. So I guess we can close it?

**Comment by tobiasd** [comment]
I'll keep an eye open to see if it still happens or not!

**Comment by Hypolite** [comment]
I've blocked most of my OStatus contacts because of this. I should try to unblock some, will report back.

**Comment by Hypolite** [comment]
Nothing to report so far!

**Comment by Hypolite** [comment]
Unfortunately, after enabling Eugen again, I got this: https://friendica.mrpetovan.com/display/52acf9cdeb9faf4c1cfb5af7e18fda5e065ba5d0

![screen shot 2017-03-03 at 11 10 14 am](https://cloud.githubusercontent.com/assets/925415/23558530/04174992-0002-11e7-9543-5d8d91c2bf91.png)

I'm not in contact with AlexJacobson, I only got the top level post because I'm following Eugen. Bonus point for the missing avatar image.

**Comment by annando** [comment]
This issue is about liked and favorited items that trigger the fetching of the whole thread. This shouldn't happen anymore.

But the thread completion when a user commented something will still be active. If you don't want this, you can deactivate the completion in total.

**Comment by Hypolite** [comment]
Ha, I found the the relevant Admin setting:

- [x] **Only import OStatus threads from our contacts**
Normally we import every content from our OStatus contacts. With this option we only store threads that are started by a contact that is known on our system.

**Comment by Hypolite** [comment]
Then I guess this issues can be closed.

**Comment by annando** [comment]
Maybe @psychohype wants to confirm that it is now working as expected.

**Comment by Hypolite** [comment]
I just got a reply from a Mastodon contact that got treated as a top level post. Is it expected?

Friendica post: https://friendica.mrpetovan.com/display/52acf9cd5a88f39044cbcecb15f8cb35497f089e
Original URL: https://mastodon.social/users/jplebreton/updates/1309403

**Comment by Hypolite** [comment]
Another example from GNU Social:

Friendica post: https://friendica.mrpetovan.com/display/f25730173ba7e0552995c1a0682d00137acd9c7c
Original URL: https://2mb.social/conversation/109598#notice-196286

**Comment by annando** [comment]
When you have disabled this "only import ..." setting from above, then - yes - this is the expected behavior.

**Comment by tobiasd** [comment]
I have set that option and get this behavior ;-) bu I thought it was a known issue and scheduled for the next version anyhow.

**Comment by Hypolite** [comment]
But the threads I pasted above have not been started by one of my contact, and I still get the reply as a top-level post somehow...

**Comment by annando** [comment]
Having the thread completion in addition to having only posts that are started by our contacts would mean a complete restructuring.

**Comment by Hypolite** [comment]
Having only posts that are started by our contacts would be good right now, as it doesn't look like it's currently happening, though.

**Comment by annando** [comment]
Problem is that the distribution of posts with OStatus is completely different from Friendica or Diaspora. They are distributed by the different servers to their own followers. So distribution speed can vary. And it could happen that you get a reply before the post. So the probability is high that you you often will have only single posts from your OStatus contacts without any comment.

**Comment by Hypolite** [comment]
Since OStatus is a high-activity, low-info per post kind of network, I'd rather have less updates but "correct" update, which means no context-less replies, and I wouldn't mind if some status updates got shafted in the process. But that just is me, I guess.

**Comment by annando** [comment]
We are having users who are happy with 10 posts per hour - While my situation is more like 10 posts per minute ;-)

**Comment by Hypolite** [comment]
Exactly! And while I expect Friendica and Diaspora to be low-activity, high-info and high reliability networks, I'm happy about the failovers and the retries, but not for OStatus.

**Comment by annando** [comment]
Question is what to do with this issue?

**Comment by annando** [comment]
@Hypolite I just realized that there is already the option "Only import OStatus threads from our contacts" in /admin/. Could you check if that works for you?

**Comment by tobiasd** [comment]
That option only work partially, as replies are shown in the timeline (and the community page as well :-( ). I've also seen "likes" in the snarl community page again.

**Comment by Hypolite** [comment]
After enabling this issue, instead of having the whole thread my OSStatus contact is answering to, I just get the reply without any context, which is worse, I disabled it.

**Comment by annando** [comment]
Could you check again?

**Comment by Hypolite** [comment]
Pulling the code right now.

**Comment by Hypolite** [comment]
Still getting a top-level OStatus thread from an unknown contact to which a known contact replied: https://friendica.mrpetovan.com/display/52acf9cd2a184d1dc81dd37b1f375e240d56d245

**Comment by annando** [comment]
Have you set the switch again to only import top level posts from your contacts?

**Comment by Hypolite** [comment]
I double-checked after seeing this post and it turns out I did.

**Comment by annando** [comment]
Okay. But maybe we should create a separate issue for this as well, since this issue here originally touched something similar but different. This issue here was that when one of my OStatus contacts had "liked" a post, then the whole thread was imported as well.

This behaviour had been changed and is working, as far as I can see.

The issue that you are reporting is that you only want to have OStatus threads in your timeline that one of our contacts has started.

These are two different things.

**Comment by Hypolite** [comment]
I opened #3018 but I was told it was a duplicate of this one. I'd gladly reopen it if it can clear things out.

**Comment by annando** [comment]
I thought you would close this one ;-)

**Comment by Hypolite** [comment]
Well, it's not mine, and I'm not sure about its status. I just know #3018 is still happening, and if this one doesn't cover it. I'll reopen mine.

**Comment by annando** [comment]
Then you should rename it, since it doesn't happen with "liking" anymore (which is why it was a duplicate)

**Comment by annando** [comment]
Can we close this one now?

**Comment by tobiasd** [comment]
I cannot remember it happening lately. But I'll keep an eye open.

**Comment by Hypolite** [comment]
Didn't happen for me in a long while 👍

**Comment by annando** [comment]
So I can close it?

**Comment by Hypolite** [comment]
Seems so.
